### DISCUSSION TITLE
Chinook Queries

### DISCUSSION THREAD

**Discussion by whoarusty** [PRQL/prql#2742] [Show and tell]
First, I just wanted to share -- I found PRQL a week ago and ... wow, what fun!

Second, during the week, I've been diving in full-steam-ahead. I see there is an open request to help out with the [Playground and Chinook queries](https://github.com/PRQL/prql/issues/1666). But, before I try to help out, I wanted to try my hand at PRQL, PRQL-Query, and DuckDB (which I also found a week ago).

So, I created a [PRQL Chinook Queries](https://github.com/whoarusty/PRQL-Chinook-Queries) repository to get things started. There are still some things I'm trying to figure out -- if I'm doing something wrong or something I should submit as an issue (mostly on the PRQL-Query side of things). I just wish I found PRQL sooner :)

If you have a moment to check out the repository, I'd be happy to hear your feedback.

Thanks for a great project,

Scott

**Comment by max-sixty** [comment]
Awesome — thanks Scott — that's looks great! Those are great examples and we could probably incorporate them somehow. Our Canonical dataset is also Chinook but the tables look slightly different — they're on the [Playground](https://prql-lang.org/playground/).

Good to see you've been using `prql-query`. That was @snth's project, I know he's spent less time on it of late but that's encouraging to see it's been helpful. We're thinking about a tool to build to make it easier to use PRQL — open to ideas & thoughts.

A couple of the issues that I see you hit we're releasing in 0.9.0 — e.g. a regex operator which will help with `like` queries. What are the other big items? Feel free to open an issue for each — some might be things we take and work on, others might be quick answers.

**Comment by snth** [comment]
Hi Scott, thanks for your feedback and setting up that repository. I will take a closer look when I get a moment. I'm glad `prql-query` has been useful for you. I have ideas for how it could do more but unfortunately haven't had the time it deserves to spend on it. I will make a concerted effort to update it after the PRQL 0.9.0 release though because there are a number of syntax changes which it won't be able to cater for otherwise.

There is also a PRQL extension for DuckDB (https://github.com/ywelsch/duckdb-prql) with which you can run PRQL right inside of DuckDB!

Please let us know any comments you have (good or bad) as you explore the language. Stumbling blocks or initial points of confusion are very important feedback and would be greatly appreciated.

Also if the rusty in @whoarusty means you are a Rust developer and want to get involved with `prql-query`, I would love any contributions, starting with updating the versions of the dependencies such as DataFusion, DuckDB or PRQL.

Looking forward to hearing more from you!

**Reply by whoarusty** [reply]
Hi Tobias,

Thanks for your response and information.

As I was working on learning PRQL I used the DuckDB PRQL extension to get the results for my queries. I then used PRQL-Query to see what the resulting SQL query would be (I don't know if the DuckDB extension allows for showing the resulting query).

The rusty in @whoarusty actually had no implications to being a Rust developer ... but maybe it could be ;) I don't currently have a Rust development environment setup ... I mostly do SQL, Python, and some C++

I'd be happy to help test PRQL and PRQL-Query so any current Windows binaries would be helpful for me.

I really like what I've been able to do with PRQL so far and look forward to seeing the project's progression.

Thanks,

Scott

**Reply by eitsupi** [reply]
> I'd be happy to help test PRQL and PRQL-Query so any current Windows binaries would be helpful for me.

Great!

[prqlc](https://github.com/PRQL/prql/tree/main/prql-compiler/prqlc) dev builds are downloadable from GitHub actions results.
https://github.com/PRQL/prql/actions/workflows/pull-request.yaml

![image](https://github.com/PRQL/prql/assets/50911393/e393a9ca-57c8-434e-80ed-2c7732c4ef6b)

**Reply by whoarusty** [reply]
I appreciate your information.

It appears I'm too inexperienced with Github ... I'm not sure how to access the release files from the Actions. I've tried looking around the links you gave and searching how to run Github Actions with no luck. Any advise for a noob?

**Reply by eitsupi** [reply]
In fact, this is quite difficult to understand if one is not familiar with GitHubActions.......
Please check this page https://docs.github.com/en/actions/managing-workflow-runs/downloading-workflow-artifacts

First, these are built with a workflow called `pull-request`, so select the name of that workflow.
![image](https://github.com/PRQL/prql/assets/50911393/a70a2f25-a8c1-4f3f-b270-1daeef3cbe2e)

Next, select the result of the execution; choose the one executed on the main branch. For example this one.
![image](https://github.com/PRQL/prql/assets/50911393/94ca3be3-a4aa-4c6b-ad49-629bf4c657f9)

https://github.com/PRQL/prql/actions/runs/5248419698

It is attached there.
![image](https://github.com/PRQL/prql/assets/50911393/06a1ddf0-726b-413a-a2a0-486dfcc33535)

**Reply by whoarusty** [reply]
🤦‍♂️ Oh, thank you for this information and screen captures. This helped so much!

I was able to get the binary file for Windows.
